cpp/vcpkg.json

```diff
@@ -31,16 +31,53 @@
       "features": [ "push", "pull" ]
     },
     "mongo-cxx-driver",
-    {
-      "name": "aws-c-io",
-      "$version reason": "To pick up https://github.com/awslabs/aws-c-io/pull/515"
-    },
     {
       "name": "aws-sdk-cpp",
-      "$version reason": "Minimum version in the baseline that works with aws-c-io above.",
+      "version>=": "1.11.405",
```

Comment:
I needed to enforce fairly new versions of all aws packages because otherwise newer versions of `aws-sdk-cpp` would not compile.

To do this I used `version>=` so in the future if aws packages fix their vcpkg dependencies we would not have to update all aws overrides when we want o upgrade the `aws-sdk-cpp`.

I think if we use an override we don't allow vcpkg to resolve dependency versions for us, but given it's currently broken either way I'll just add an override only for `aws-sdk-cpp`. I'll leave the other aws packages with `version>=` to not require upgrading all overrides when we only want to upgrade the `aws-sdk-cpp`